### PR TITLE
MMD OD/IS sdk, cli examples; Update HuggingFace Classification examples

### PR DESCRIPTION
**Pull request by rjaincc**
# Description
* Update Classification examples with latest updates
* Introduce MMD OD and IS sdk and cli examples. 

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/foundation-models/system/finetune/image-classification/multilabel-classification/hftransformers-fridgeobjects-multilabel-classification-pipeline.yaml [hunk 1]
```diff
@@ -58,7 +58,6 @@ jobs:
       training_data: ${{parent.inputs.training_data}}
       validation_data: ${{parent.inputs.validation_data}}
 
-      auto_hyperparameter_selection: False
```
Comment (MadhuM02) [inline cli/foundation-models/system/finetune/image-classification/multilabel-classification/hftransformers-fridgeobjects-multilabel-classification-pipeline.yaml:61]: Should we also remove the params which are picked up from defaults??

File: sdk/python/foundation-models/system/finetune/image-object-detection/mmdetection-fridgeobjects-object-detection.ipynb [hunk 2]
```diff
@@ -0,0 +1,1123 @@
...
+    "\n",
+    "try:\n",
+    "    _ = workspace_ml_client.compute.get(finetune_cluster_name)\n",
+    "    print(\"Found existing compute target.\")\n",
+    "except ResourceNotFoundError:\n",
+    "    print(\"Creating a new compute target...\")\n",
+    "    compute_config = AmlCompute(\n",
+    "        name=finetune_cluster_name,\n",
+    "        type=\"amlcompute\",\n",
+    "        size=\"Standard_NC6\",\n",
```
Comment (MadhuM02) [inline sdk/python/foundation-models/system/finetune/image-object-detection/mmdetection-fridgeobjects-object-detection.ipynb:184]: change this to Standard_NC6s_v3??

File: cli/foundation-models/system/finetune/image-instance-segmentation/mmdetection-fridgeobjects-instance-segmentation.sh [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+# Scoring file
+mmdetection_sample_request_data="./mmdetection_sample_request_data.json"
+
+# finetuning job parameters
+# TODO: update with preview registry component name
+finetuning_pipeline_component="mmdetection_image_objectdetection_instancesegmentation_pipeline"
+
+# Training settings
+number_of_gpu_to_use_finetuning=$gpus_per_node # set to the number of GPUs available in the compute
```
Comment (MadhuM02) [inline cli/foundation-models/system/finetune/image-instance-segmentation/mmdetection-fridgeobjects-instance-segmentation.sh:44]: number_of_gpu_to_use_finetuning -> process_count_per_instance??

File: cli/foundation-models/system/finetune/image-instance-segmentation/mmdetection-fridgeobjects-instance-segmentation-pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      # finetuning parameters
+      apply_augmentations: True
+      image_min_size: -1
+      image_max_size: -1
+      metric_for_best_model: mean_average_precision
+      number_of_workers: 8
+      apply_deepspeed: False
+      deepspeed_config: ${{parent.inputs.ds_finetune}}
+      apply_ort: False
+      number_of_epochs: 15
```
Comment (gauravrajguru) [inline cli/foundation-models/system/finetune/image-instance-segmentation/mmdetection-fridgeobjects-instance-segmentation-pipeline.yml:75]: do we need to remove HP values if auto selection is on?

File: sdk/python/foundation-models/system/finetune/image-instance-segmentation/mmdetection-fridgeobjects-instance-segmentation.ipynb [hunk 5]
```diff
@@ -0,0 +1,1062 @@
...
+    "### 3. Pick a foundation model to fine tune\n",
+    "\n",
+    "We will use the `mask_rcnn_r50_fpn_1x_coco` model in this notebook. If you need to fine tune a model that is available on MMDetection model zoo, but not available in `azureml` system registry, you can either register the model and use the registered model or use the `model_name` parameter to instruct the components to pull the model directly from MMDetection model zoo.\n",
+    "\n",
+    "Currently following models are supported:\n",
+    "\n",
+    "| Model Name | Source |\n",
+    "| :------------: | :-------:  |\n",
+    "| [Image instance-segmentation models from MMDetection](https://github.com/open-mmlab/mmdetection/blob/main/docs/en/model_zoo.md) | MMDetection |\n",
+    "| TODO | azureml-preview registry |"
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/finetune/image-instance-segmentation/mmdetection-fridgeobjects-instance-segmentation.ipynb:206]: can be updated

**Review by gauravrajguru — CHANGES_REQUESTED** [review]

File: sdk/python/foundation-models/system/finetune/image-classification/multiclass-classification/hftransformers-fridgeobjects-multiclass-classification.ipynb [hunk 6]
```diff
@@ -498,7 +498,6 @@
     "    \"mlflow_model\": foundation_model,\n",
     "    # \"model_name\": huggingface_model_name,\n",
     "    # finetune_args\n",
-    "    \"auto_hyperparameter_selection\": False,\n",
```
[the remark this answers is not in the document]
Comment (rjaincc) [reply sdk/python/foundation-models/system/finetune/image-classification/multiclass-classification/hftransformers-fridgeobjects-multiclass-classification.ipynb:501]: We are not. This parameter has been taken out.

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_finetune_component.md [hunk 7]
```diff
@@ -15,35 +14,28 @@ The components can be seen here ![as shown in the figure](../../images/image_cla
 
     Path to the mltable folder of validation dataset.
 
-4. _auto_hyperparameter_selection_ (bool, optional)
```
[the remark this answers is not in the document]
Comment (rjaincc) [reply sdk/python/foundation-models/system/docs/component_docs/image_finetune/transformers_finetune_component.md:18]: We don't have it anymore. No defaults on the UI, anything that user mentioned will be picked else will come from hierarchy.

File: sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
+# MMDetection Model Finetune Component
+This component enables finetuning of pretrained models on custom or pre-available datasets. The component supports Deepspeed and ONNXRuntime configurations for performance enhancement. 
+The components can be seen here ![as shown in the figure](../../images/image_mmd_od_is_finetune_components.jpg)
+
+# 1. Inputs
+1. _model_path_ (URI_FOLDER, required)
```
[the remark this answers is not in the document]
Comment (rjaincc) [reply sdk/python/foundation-models/system/docs/component_docs/image_finetune/mmd_finetune_component.md:6]: We don't have it anymore.

**Review by gauravrajguru — APPROVED** [review]

**Pull request merged**